feat(taskfiles)!: Add python script to replace confusing shell find usage in tasks. (#59)
y-scope/yscope-dev-utils: exports/ystdlib-py/README.md

# ystdlib-py

Python utilities developed and used at YScope.

> ## Modules

Reviewer: ```suggestion
## Usage
```

> How to get started with the modules inside ystdlib-py.
> 

Reviewer: ```suggestion
```

Although we shouldn't have two consecutive headings, I think this sentence is a bit redundant.

### find.py

> List all Python files exported from yscope-dev-utils, but excluding anything from a virtual env:

Reviewer: ```suggestion
Example: *List all Python files exported from yscope-dev-utils, excluding anything from a virtual
environment.*
```

```shell
uv run find yscope-dev-utils --include "**/exports/**" --exclude "**/.venv/**" --filename "*.py"
```

Display detailed information for the command:

```shell
uv run find --help

> ```

Reviewer: Can we put this first?

## Contributing

Before you submit a pull request, ensure you follow the testing and linting instructions below.

### Requirements

* [Task] 3.40 or higher
* [uv] 0.7.10 or higher

### Set up

To install all Python dependencies run:

```shell
uv sync
```

### Testing

Testing is run through the yscope-dev-utils taskfile. To run all tests:

```shell
task tests:ystdlib-py
```

> To see how to run a subset of tests run:

Reviewer: ```suggestion
To run a subset of tests, list all tests:
```

```shell
task -a
```

Look for all tasks under the `tests` namespace (identified by the `tests:` prefix).

### Linting

> Before submitting a pull request, ensure you’ve run the linting commands below and have fixed all

Reviewer: ```suggestion
Before submitting a pull request, ensure you've run the linting commands below and have fixed all
```
